packages/dds/tree/src/feature-libraries/chunked-forest/codec/schemaBasedEncoding.ts

```diff
@@ -64,12 +72,24 @@ export function fieldShaper(
 	treeHandler: TreeShaper,
 	field: TreeFieldStoredSchema,
 	cache: EncoderCache,
+	storedSchema: StoredSchemaCollection,
 ): FieldEncoder {
 	const kind = cache.fieldShapes.get(field.kind) ?? fail("missing FieldKind");
 	const type = oneFromSet(field.types);
 	const nodeEncoder = type !== undefined ? treeHandler.shapeFromTree(type) : anyNodeEncoder;
-	// eslint-disable-next-line unicorn/prefer-ternary
 	if (kind.multiplicity === Multiplicity.Single) {
+		if (field.kind === identifierFieldKindIdentifier) {
+			assert(type !== undefined, "field type must be defined in identifier field");
+			const nodeSchema = storedSchema.nodeSchema.get(type);
+			assert(nodeSchema !== undefined, "nodeSchema must be defined");
+			assert(
+				nodeSchema instanceof LeafNodeStoredSchema,
+				"nodeSchema must be LeafNodeStoredSchema",
+			);
+			assert(nodeSchema.leafValue === 1, "identifier field can only be type string");
```

Comment:
is there not a more intentional way to assert this (a named enum, perhaps)? or is comparison to 1 the best we have?

Reply:
Updated to assert against `ValueSchema.String`